fair/2.0.0/emissions_driven.py

```diff
@@ -0,0 +1,186 @@
+import pandas as pd
+import numpy as np
+from tools import unifiedtools
+
+def run(inp_df, cfg):
+    """
+    Run FaIR 2.0
+
+    Parameters
+    ----------
+    inp_df : :obj:`pd.DataFrame`
+        Input :obj:`pd.DataFrame` containing the timeseries to run
+
+    cfg : dict
+        Dictionary containing the configuration for this run
+
+    Returns
+    -------
+    :obj:`pd.DataFrame`
+        Results of the run
+    """
+    np_input = _convert_df_to_numpy(inp_df)
+    #res_numpy = _run_numpy(np_input, cfg)
+    #res = _convert_numpy_output_to_df(res_numpy)
+    res = pd.DataFrame()
+    return res
+
+def _run_numpy(inp_ar, 
+                a1,
+                a2,
+                a3,
+                a4,
+                tau1,
+                tau2,
+                tau3,
+                tau4,
+                r0,
+                rC,
+                rT,
+                rA,
+                PI_conc,
+                emis2conc,
+                f1,
+                f2,
+                f3,
+                d,
+                q,
+                ext_forcing,
+                timestep):
+    """
+    Run FaIR 2.0 from numpy array
+
+    This function is internal/private (because its name starts with an
+    underscore) i.e. not intended for use by people other than developers.
+    I wouldn't recommend having a 'public' interface which is just based on
+    numpy as it's too easy to get column orders etc. confused so it'll create
+    more problems than its worth.
+
+    This function can *only* run one scenario, thermal parameter set & gas parameter set at a time
+
+    Parameters
+    ----------
+    inp_ar : :obj:`np.ndarray`
+        Input :obj:`np.ndarray` containing the timeseries to run. No checks
+        of the column order are performed here.
+        format: [[species],[time]]
+
+    a1,...,f3 : :obj:`np.ndarray`
+        Input :obj:`np.ndarray` containing gas parameters in format: [species], 
+        note: all species contain the same number of gas/thermal pool indices (some are simply populated with 0)
+
+    d,q : obj:`np.ndarray`
+        Input :obj:`np.ndarray` containing thermal parameters in format: [response box]
+    
+    ext_forcing : :obj:`np.ndarray`
+        Input :obj:`np.ndarray` containing aggregated external forcing in format: [time]
+    
+    timestep : :obj:`np.ndarray`
+        Input :obj:`np.ndarray` specifying each timestep of inp_ar
+    
+
+    Returns
+    -------
+    :obj:`np.ndarray`
+        Results of the run. Given in the format: [[Concentration],[RF],[T],[alpha],[Emissions],[gas parameters],[thermal parameters]]
```

Comment:
The format here doesn't make sense to me. Is this an array with dimensions for concentration, RF, T etc. (like the above format would imply e.g. a format of [[species], [time]] means the first dimension is different species, the second is different times) or is this a list of returns, each of which has its own dimensions e.g. Concentration has dimensions of [[species], [time]]?

Reply:
Having just read the code, the output is actually a dictionary, can be described as shown below (more info on numpy docstrings is here https://numpy.org/devdocs/docs/howto_document.html)

```
Returns
--------
dict
    Dictionary containing the results. Keys are [description of keys]. Values are [description of values]
```

Reply:
Done